File: stack from queue.cpp
#include <stdio.h>

class queue{
    int *arr ;

public:
    int front = 0;
    int rear = -1;
    int size ;
    /*constructor */
    explicit queue(int size ){
        this->size = size;
        arr = new int[size];
    }
    queue(){
        this->size = 100;
        arr = new int[size];
    }
    void enQueue(int data){
        if (rear < size - 1){
            rear++;
            arr[rear] = data;
        }
        else{
            printf("%s","queue is full");
        }
    }
    int deQueue(){
        if (rear == -1){
            printf("%s","queue is empty");
        }
        else{
            int temp = front;
            front++;
            if (front > rear){
                front =  rear = -1;
            }
            return arr[temp];
        }
    }
    void display(){
        if (rear == -1){
            printf("%s","queue is empty");
        }
        int count =1;
        for (int i = front; i <= rear ; ++i) {
            printf("data in %d container is %d\n",count,arr[i]);
            count++;
        }
    }
};

class stackFromQueue {
    queue q1;
    int top = -1;
public:
    void push(int data){
        q1.enQueue(data);
        top++;
    }
    void pop(){
        queue q2;
        for (int i = 0; i < top; ++i) {
            q2.enQueue(q1.deQueue());
        }
        q1 = q2;
        top--;
    }
    void peek(){
        q1.display();
    }
};

int main(){
    stackFromQueue s1;
    s1.push(10);
    s1.push(11);
    s1.push(12);
    s1.push(13);
    s1.peek();
    s1.pop();
    s1.peek();

    return  0;
}

File: stack from queue.py
class queue:
    def __init__(self):
        self.listq = []

    def enQueue(self, data):
        self.listq.append(data)

    def deQueue(self):
        if not self.listq:
            print("queue is empty ")
        else:
            temp = self.listq[0]
            self.listq.pop(0)
            return temp

    def display(self):
        if not self.listq:
            print("queue is empty ")
        else:
            count = 1;
            for i in range(len(self.listq)):
                print("data in pos  {0} is {1}".format(count, self.listq[i]))
                count = count + 1


class stackFromQueue():
    def __init__(self):
        self.top = -1
        self.q1 = queue()

    def push(self, data):
        self.q1.enQueue(data)
        self.top = self.top + 1

    def pop(self):
        q2 = queue()
        for i in range(self.top ):
            q2.enQueue(self.q1.deQueue())
        self.q1 = q2
        self.top = self.top - 1

    def peek(self):
        self.q1.display()


if __name__ == '__main__':
    s1 = stackFromQueue()
    s1.push(10)
    s1.push(11)
    s1.push(12)
    s1.peek()
    s1.pop()
    s1.peek()
